### ISSUE TITLE
Running AutoML simultaneously result in fatal Error

### ISSUE THREAD

**Issue by schwannden** [h2oai/sparkling-water#532]
Given the following setup:
```scala
def train(frame: H2OFrame, response: String)(implicit h2OContext: H2OContext): AutoML = {
  val autoSpec = new AutoMLBuildSpec()
  autoSpec.input_spec.training_frame = frame.key
  autoSpec.input_spec.response_column = response
  autoSpec.build_control.stopping_criteria.set_seed(777)
  autoSpec.build_control.stopping_criteria.set_max_models(10)
  autoSpec.build_control.stopping_criteria.set_max_runtime_secs(30)
  autoSpec.build_control.stopping_criteria.set_stopping_metric(StoppingMetric.misclassification)
  autoSpec.build_control.project_name = s"Auto_$frame"
  autoSpec.build_control.nfolds = 3

  val aml = AutoML.makeAutoML(Key.make(), new Date(), autoSpec)
  AutoML.startAutoML(aml)
  aml
}

val frame1 = new H2OFrame(URI.create("data/iris.csv"))
val frame2 = new H2OFrame(URI.create("data/iris2.csv"))
```
The following code fails:
```scala
List(frame1, frame2).map {frame =>
  train(frame, "class")
} foreach (_.get())
```
But the following code runs successfully:
```scala
List(frame1, frame2).map {frame =>
  train(frame, "class").get()
}
```

So it seems like we are not able to run 2 AutoML at the same time? The error message is:
```
Name lookup of 'keyW' failed
java.lang.IllegalArgumentException: Name lookup of 'keyW' failed
	at water.rapids.Env.lookup(Env.java:402)
	at water.rapids.ast.params.AstId.exec(AstId.java:33)
	at water.rapids.ast.prims.advmath.AstKurtosis.apply(AstKurtosis.java:24)
	at water.rapids.ast.prims.advmath.AstKurtosis.apply(AstKurtosis.java:10)
	at water.rapids.ast.AstExec.exec(AstExec.java:63)
	at water.rapids.Session.exec(Session.java:88)
	at water.rapids.Rapids.exec(Rapids.java:64)
	at ai.h2o.automl.FrameMetadata$MetaPass1.<init>(FrameMetadata.java:695)
	at ai.h2o.automl.FrameMetadata.computeFrameMetaPass1(FrameMetadata.java:571)
	at ai.h2o.automl.AutoML.learn(AutoML.java:1063)
	at ai.h2o.automl.AutoML.run(AutoML.java:439)
	at ai.h2o.automl.H2OJob$1.compute2(H2OJob.java:32)
	at water.H2O$H2OCountedCompleter.compute(H2O.java:1263)
	at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
	at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
	at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
	at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
	at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)
```

**Comment by schwannden** [comment]
It is likely due to duplicated naming in grid search for the two AutoML algorithm. Is there a way to configure AutoMLSpec so that grid search append a different prefix to its internal models?

**Comment by jakubhava** [comment]
Hi @schwannden, thanks for the bug report. Let me include the authors of AutoML and see if they can help. If this would be related just to sparkling water environment, I will go ahead and try it out.

CC: @rpeck @ledell

**Comment by mmalohlava** [comment]
@schwannden thanks for report! It seems like internal Rapid session is not properly initialized.

**Comment by michalkurka** [comment]
This is a bug in H2O, there is a a piece of code that uses hardcoded Key name ("keyW"). This is what is causing the conflict.

**Comment by michalkurka** [comment]
Will be fixed in https://0xdata.atlassian.net/browse/PUBDEV-5228, target release 3.18.0.1

**Comment by jakubhava** [comment]
@schwannden I'm closing this issue for now. Please use the link to the Jira to track the progress. 

Thanks again for reporting!

**Comment by michalkurka** [comment]
Fixed in H2O (will be in 3.18.0.1)
